lib/proxy/router.go

```diff
@@ -147,7 +204,12 @@ func (r *Router) DialHost(ctx context.Context, from net.Addr, host, port, cluste
 			attribute.String("site", clusterName),
 		),
 	)
-	defer span.End()
+	defer func() {
+		span.End()
+		if err != nil {
+			failedConnectingToNode.Inc()
+		}
```

Comment:
Technically
```suggestion
		if err != nil {
			failedConnectingToNode.Inc()
		}
		span.End()
```
or maybe the error check should be in a separate defer that runs before `span.End`